builtInFunctions/esdtNFTTransfer.go

```diff
@@ -419,6 +419,12 @@ func (e *esdtNFTTransfer) addNFTToDestination(
 
 	transferValue := big.NewInt(0).Set(esdtDataToTransfer.Value)
 	esdtDataToTransfer.Value.Add(esdtDataToTransfer.Value, currentESDTData.Value)
+
+	err = e.esdtStorageHandler.RemoveNFTMetadataFromSystemAccountIfNeeded(esdtTokenKey, nonce, currentESDTData)
```

Comment:
can't we do this under SaveESDTNFTToken ? as it is called always before that.

Reply:
Moved it there.